python/samples/concepts/embedding/text_embedding_generation.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+
+from numpy import dot, linalg
+
+from samples.concepts.setup.text_embedding_services import Services, get_text_embedding_service_and_request_settings
+
+"""
+This sample shows how to generating embeddings for text data. This sample uses the following component:
+- an text embedding generator: This component is responsible for generating embeddings for text data.
+"""
+
+
+# You can select from the following text embedding services:
+# - Services.OPENAI
+# - Services.AZURE_OPENAI
+# - Services.AZURE_AI_INFERENCE
+# - Services.BEDROCK
+# - Services.GOOGLE_AI
+# - Services.HUGGING_FACE
+# - Services.MISTRAL_AI
+# - Services.OLLAMA
+# - Services.VERTEX_AI
+# Please make sure you have configured your environment correctly for the selected text embedding service.
+text_embedding_service, request_settings = get_text_embedding_service_and_request_settings(Services.OPENAI)
+
+TEXTS = [
+    "A dog ran joyfully through the green field, chasing after butterflies in the warm afternoon sun.",
+    "A happy puppy sprinted across the grassy meadow, playfully pursuing insects under the bright sky.",
+]
+
+
+def cosine_similarity(a, b):
```

Comment:
You can use: `from scipy.spatial.distance import cosine` this is installed by default in SK

Reply:
and then do `1 - the outcome` for cosine similarity (cosine == cosine_distance)